sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -128,6 +128,172 @@ case class MapKeys(child: Expression)
   override def prettyName: String = "map_keys"
 }
 
+@ExpressionDescription(
+  usage = """
+    _FUNC_(a1, a2, ...) - Returns a merged array of structs in which the N-th struct contains all
+    N-th values of input arrays.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 2, 3), array(2, 3, 4));
+        [[1, 2], [2, 3], [3, 4]]
+      > SELECT _FUNC_(array(1, 2), array(2, 3), array(3, 4));
+        [[1, 2, 3], [2, 3, 4]]
+  """,
+  since = "2.4.0")
+case class ArraysZip(children: Seq[Expression]) extends Expression with ExpectsInputTypes {
+
+  override def inputTypes: Seq[AbstractDataType] = Seq.fill(children.length)(ArrayType)
+
+  override def dataType: DataType = ArrayType(mountSchema)
```

Comment:
Just a quick follow up question... Under what circumstances does the output array can contain null elements? Shouldn't the output dataType be ```ArrayType(mountSchema, false)```?

Reply:
In the first test case I zipped ``Seq(9001, 9002, 9003, null)`` with ``Seq(null, 1L, null, 4L, 11L)``, and expected the result to be ``Seq(Seq(9001, null), Seq(9002, 1L), ..., Seq(null, 11L))``, for instance.

I tried to define the nullability (this word exist? haha) of the output in runtime, but I thought that it was not possible since I can't eval every result before defining the schema.

What you think?

Reply:
I understand that fields of the nested ```struct``` can be ```null```, but can you give me an example of the input that would lead to something like ```Seq(null, Seq(9002, 1L))```?

Reply:
Hmmm you are correct then, I don't think that such scenario could happen (correctly, at least). That means that the dataType should always reject null values?

Reply:
Yeah, seems like the struct which is the element of the array is not null, so the data type would be `ArrayType(mountSchema, containsNull = false)`.

Reply:
Ok, I will fix it as a part of #21352.

Reply:
the struct can be null if any of the input element is null IIUC. So probably `ArrayType(mountSchema, containsNull = children.exists(_nullable))`?

Reply:
In that case, the array itself will be null and `nullable` is already `children.exists(_nullable)`.

Reply:
yes, you're right, sorry!